### PR TITLE
support for explicit test_dataset definition for evals

### PR DESCRIPTION
**Pull request by winglian**
first pass for supporting different datasets for evals rather than splitting the test dataset.

### CODE REVIEW

File: src/axolotl/utils/data.py [hunk 1]
```diff
@@ -239,8 +251,9 @@ def for_d_in_datasets(dataset_configs):
                 raise ValueError("unhandled dataset load")
             # support for using a subset of the data
             if config_dataset.shards:
-                if "train" in ds:
-                    ds = ds.shuffle(seed=seed)["train"].shard(
+                shard_split = config_dataset.split if config_dataset.split else split
```
Comment (NanoCode012) [inline src/axolotl/utils/data.py]: Maybe we can refactor this code to split first (using the code around L270-275), then shard after?

**Review by NanoCode012 — COMMENTED** [review]
Hmm, our data pipeline is getting a bit complex. I had a bit of hard time following the flow

**Review by NanoCode012 — COMMENTED** [review]
Adding this config to Readme would be helpful

**Comment by winglian** [comment]
This will address #875

**Comment by NanoCode012** [comment]
I think will need to add to doc about this and also, whether it would be appropriate to hardcode the `train`/`test` split or allow more open control?

**Comment by winglian** [comment]
> I think will need to add to doc about this and also, whether it would be appropriate to hardcode the `train`/`test` split or allow more open control?

@NanoCode012 What do you mean about hardcoding?

**Comment by DreamGenX** [comment]
Do you have an example or documentation of how this can be used?

**Comment by DreamGenX** [comment]
I was trying to reserve engineer how it's supposed to work, and maybe there's a bug here:

https://github.com/OpenAccess-AI-Collective/axolotl/blob/18f811978c01d567c2294140f53abcf8c086e337/src/axolotl/utils/data.py#L443

```py
    dataset, prompters = load_tokenized_prepared_datasets(
        tokenizer, cfg, default_dataset_prepared_path
    )
```

Shouldn't you pass `split=split` to `load_tokenized_prepared_datasets`? Otherwise it will never load from `cfg.test_datasets`.

**Pull request merged**